src/games.rb

```diff
@@ -36,18 +47,23 @@ def read_file
       end.slice_after(/InitGame/)
     end
   
-    def find_match
+    def find_match(regex)
       game = read_file.map do |gamelines|
         gamelines.map do |gameline|
-          match = gameline.match(/:(?<killer>[^:]+) killed (?<killed>.+) by/)
+          match = gameline.match(regex)
           match ? match.named_captures : nil
         end.compact
      
       end
-    #   game.delete([])
       game
     end
-
+    
+    def count_deaths(match)
+      match_count = match.group_by(&:itself).to_h do |death, means_of_death| 
+         [death.values, means_of_death.count].sort_by{ |key, value| value }
+       end
+       match_count.sort_by{ |key, value| value }.reverse.to_h
```

Comment:
why are you sorting here and in line 63?
that sort in line 63 can probably be removed.